gef.py

```diff
@@ -8907,22 +8930,31 @@ def do_invoke(self, argv: List[str]) -> None:
         headers = ["Start", "End", "Offset", "Perm", "Path"]
         gef_print(Color.colorify("{:<{w}s}{:<{w}s}{:<{w}s}{:<4s} {:s}".format(*headers, w=gef.arch.ptrsize*2+3), color))
 
+        last_printed_filter = None
+
         for entry in vmmap:
-            if not argv:
+            names_filter = [f"name = '{x}'" for x in names if x in entry.path]
+            addrs_filter = [f"addr = {self.format_addr_filter(arg, addr)}" for arg, addr in addrs.items()
+                if entry.page_start <= addr < entry.page_end]
+            filters = names_filter + addrs_filter
+            filter_content = f"[{' & '.join(filters)}]"
+
+            if len(names) + len(addrs) == 0:
                 self.print_entry(entry)
```

Comment:
I think you could continue after this, and then remove the `elif` below, & un-indent the rest. Also, couldn't the `if` above just be `if not filter_content`?

Reply:
We can't remove the elif because we want to check if there are matched filters (which is not already checked)
And we can't replace the first if with `if not filter_content` since we want to check if there are existing filters, but this would check if there are matched filters

Reply:
ok, then I think `if not names and not addrs` is probably clearer

Reply:
up to you though. Approved.